File: kotlin/basics/FizzBuzz.kt
// Print FizzBuzz for 1 through 100, choosing the label with a when expression.
// Run: kotlinc FizzBuzz.kt -include-runtime -d fizzbuzz.jar && java -jar fizzbuzz.jar

fun fizzBuzzLabel(n: Int): String = when {
    n % 15 == 0 -> "FizzBuzz"
    n % 3 == 0 -> "Fizz"
    n % 5 == 0 -> "Buzz"
    else -> n.toString()
}

fun main() {
    for (n in 1..100) {
        println(fizzBuzzLabel(n))
    }
}

File: dart/basics/fizzbuzz.dart
// Print FizzBuzz for 1 through 100, choosing the label with a switch expression on a record.
// Run: dart run fizzbuzz.dart

String fizzBuzzLabel(int n) => switch ((n % 3, n % 5)) {
      (0, 0) => 'FizzBuzz',
      (0, _) => 'Fizz',
      (_, 0) => 'Buzz',
      _ => '$n',
    };

void main() {
  for (var n = 1; n <= 100; n++) {
    print(fizzBuzzLabel(n));
  }
}
